bootstrap/src/sun/nio/ch/lincheck/Injections.java

```diff
@@ -370,7 +373,16 @@ public static void afterWrite() {
      * @return Deterministic call descriptor or null.
      */
     public static Object onMethodCall(Object owner, String className, String methodName, int codeLocation, int methodId, String methodDesc, Object[] params) {
-        return getEventTracker().onMethodCall(owner, className, methodName, codeLocation, methodId, methodDesc, params);
+        // to safely construct the method signature we need to enter ignored section
+        // because it internally calls code which has instrumentation
+        enterIgnoredSection();
+        MethodSignature methodSignature;
+        try {
+            methodSignature = new MethodSignature(methodName, convertAsmMethodType(methodDesc));
+        } finally {
+            leaveIgnoredSection();
+        }
```

Comment:
Implementing a method, which would accept `Runnable`/`Supplier<T>` is not as straightforward: on old jdks (11, 13), calling a method with passed lambda results in infinite `Injections::onMethodCall` invocation for method `java/lang.invoke.DirectMethodHandle::internalMethodName`. Which might be related to lambda capturing some outer variables